### ISSUE TITLE
Cannot connect to the i3 window manager when started before i3

### ISSUE THREAD

**Issue by Problembeloeser** [denesb/xfce4-i3-workspaces-plugin#62]
I know I got a bit of a strange setup, but let me explain what I think is relevant:

I originally used Arch with xfce and recently followed [this guide from Feeble Nerd](http://feeblenerd.blogspot.com/2015/11/pretty-i3-with-xfce.html) to replace xfwm with i3. Now I installed and added the workspaces plugin and added it to the xfce panel. Now the issue: When logging in, the panel is initialized before i3 is started, causing this error. However, as the CPU is spammed with those requests, i3 will not boot up, so this essentially prevents me from starting into xfce, as it blocks anything from starting. I can fix it by removing the workspaces plugin from the xfce panel config xml, but this is only a workaround.

The following line is spammed into `.xsession-errors` causing it to inflate to 200 MiB within a couple seconds:

```Maximum number of clients reachedMaximum number of clients reachedCannot connect to the i3 window manager: socket path atom reply is null```.

For the time being, I cannot use this plugin properly. If there is any workaround possible, please tell me, I really want to get this working somehow.

**Comment by Kwbmm** [comment]
I believe I have the same issue and I am experiencing the same as #56 . I think these 2 issues are the same.
I don't have a solution unfortunately... I'm thinking about a hack to load the plugin at startup and unload it at shutdown, but it might be tricky to pull off

**Comment by Kwbmm** [comment]
I might have found a workaround.

Looking at the [Arch wiki](https://wiki.archlinux.org/index.php/Xfce#Use_a_different_window_manager), I ran the `xfconf-query` to change the default window manager:

```sh
$ xfconf-query -c xfce4-session -p /sessions/Failsafe/Client0_Command -t string -sa xfsettingsd
$ xfconf-query -c xfce4-session -p /sessions/Failsafe/Client1_Command -t string -sa i3
```
Once I ran the above command, I looked at what `xfce4-settings-editor` was showing for the xfce4-session in terms of priorities:
![image](https://user-images.githubusercontent.com/8115374/114322345-03cda680-9b20-11eb-9de4-5b3517c70c8b.png)

This basically meant that I told the system to run first `i3` and then `xfce4-panel`. Doing so, the plugin doesn't block, because `i3` is already started.

**Comment by jeremyjjbrown** [comment]
I have a similar issue. Trying to quit the panel from a remote session yeilds the same error. I was forced to logout and remove .config/xfce from a shell and reconfigure my session.

**Comment by hartlottery** [comment]
Although the issue is quite "old", but we have modified the code to address it, using the timer technique. Pull request is in #70, still under testing in my daily archlinux gear, any feedback is welcomed :)

**Comment by denesb** [comment]
Closing this as it should be fixed by #70. Let's reopen if still seen.
